sdk/trace/src/main/java/io/opentelemetry/sdk/trace/AndroidFriendlyRandomIdGenerator.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.api.trace.SpanId;
+import io.opentelemetry.api.trace.TraceId;
+import java.util.Random;
+
+/**
+ * {@link IdGenerator} instance that doesn't use {@link java.util.concurrent.ThreadLocalRandom},
+ * which is broken on most versions of Android (it uses the same seed everytime it starts up).
+ */
+enum AndroidFriendlyRandomIdGenerator implements IdGenerator {
+  INSTANCE;
+
+  private static final ThreadLocal<Random> threadLocalRandom = new ThreadLocal<>();
+  private static final long INVALID_ID = 0;
+
+  @Override
+  public String generateSpanId() {
+    long id;
+    Random random = getRandom();
+    do {
+      id = random.nextLong();
+    } while (id == INVALID_ID);
+    return SpanId.fromLong(id);
+  }
+
+  @Override
+  public String generateTraceId() {
+    long idHi;
+    long idLo;
+    Random random = getRandom();
+    do {
+      idHi = random.nextLong();
+      idLo = random.nextLong();
+    } while (idHi == INVALID_ID && idLo == INVALID_ID);
+    return TraceId.fromLongs(idHi, idLo);
+  }
+
+  private static Random getRandom() {
```

Comment:
This can be replaced with ThreadLocal.initialValue() (https://docs.oracle.com/javase/7/docs/api/java/lang/ThreadLocal.html#initialValue()) which was intended for this kind of initialisation.

Reply:
I knew there was an API for this...I looked and couldn't find it, but obviously I missed what was staring me in the face. BTW, there's an even easier way now:

`ThreadLocal.withInitial(Random::new)`

Reply:
Of course, that can't be used on Android API 21. 🤦🏼

Reply:
Nice, love the supplier version. Easy way to make this whole class generic, ie allowing supplier to be injected via constructor. However now probably an overkill due to lack of useful use-cases for that.

Reply:
I did consider that as well, but this is a small enough amount of code that I think it's fine. Since this is an internal API, we can always change it in the future if we need a more flexible id generator setup.

Reply:
Update to use the `initialValue()` override.